modules/core/src/webgl/classes/query.js

```diff
@@ -139,7 +141,15 @@ export default class Query extends Resource {
 
   // Returns true if the query result is available
   isResultAvailable() {
-    return this.gl.getQueryParameter(this.handle, GL_QUERY_RESULT_AVAILABLE);
+    const resultAvailable = this.gl.getQueryParameter(this.handle, GL_QUERY_RESULT_AVAILABLE);
+    if (resultAvailable) {
+      this.queryPending = false;
+    }
+    return resultAvailable;
+  }
+
+  isTimerDisjoint() {
+    return this.gl.getParameter(GL_GPU_DISJOINT_EXT);
```

Comment:
I don't recall the details, but doesn't the `QueryManager` class also handle GL_GPU_DISJOINT_EXT? Worth a check so we don't handle it twice...

The API seems a little clunky for the user. 
* `isResultAvailable()` could check queryPending and return false if not pending.
* It could perhaps also clear `queryPending` flag and return false if disjoint?

Perhaps it would be taking too much liberty with the WebGL -> luma.gl mapping but I envision something like:

Since this pattern is necessary to deal with the lag in completion of timers, we should probably add such a code example to the docs, regardless of how we settle on the API.

```
_beginTimers() {
    if (this.gpuTimeQuery) {
      if (!this.gpuTimeQuery.isQueryPending()) {
        this.gpuTimeQuery.beginTimeElapsedQuery();
      } else if (this.gpuTimeQuery.isResultAvailable()) {
        this.gpuTime = this.gpuTimeQuery.getResult();
        this.stats.addTime("GPU Time", this.gpuTime);
      }
    }

    this._cpuStartTime = getHiResTimestamp();
  }

  _endTimers() {
    this.cpuTime = getHiResTimestamp() - this._cpuStartTime;
    this.stats.addTime("CPU Time", this.cpuTime);

    if (this.gpuTimeQuery && !this.gpuTimeQuery.queryPending) {
      this.gpuTimeQuery.end();
    }
  }

Reply:
There's [this code](https://github.com/uber/luma.gl/blob/master/modules/core/src/webgl/classes/query.js#L169-L174) in query.js class, but I believe that's to invalidate asynchronous queries?

You reminded me of a change I need to make that requires exposing the disjoint timer status: I want to set the gpuTime to -1 to indicate that the result is not valid (e.g. so it doesn't get accumulated like I'm doing in the examples). You're right that this will require some documentation.

We can't do it quite how you presented, because getting the results should be independent of beginning the query, so that if results are available we can immediately begin the next query. I like the idea of isResultAvailable checking the queryPendingFlag though. That will simplify this code a bit. I'll add some comments explaining the structure as well.